### ISSUE TITLE
Game crashes with failed AI assertion

### ISSUE THREAD

**Issue by vlk-charles** [ihhub/fheroes2#7650]
### Preliminary checks

- [X] I've checked that there aren't [**other open issues**](https://github.com/ihhub/fheroes2/issues?q=is%3Aissue) on the same topic.
- [X] I've checked that this issue is reproducible on the [**latest snapshot build**](https://ihhub.github.io/fheroes2/INSTALL.html#snapshots-latest-builds).

### Platform

Linux

### Describe the bug

Game crashes after loading the attached save file and ending turn.

### Save file

[AI_crash.sav.zip](https://github.com/ihhub/fheroes2/files/12449024/AI_crash.sav.zip)

### Additional info

Console output:
```
fheroes2: ../fheroes2/ai/normal/ai_normal_hero.cpp:2192: void AI::Normal::updatePriorityTargets(Heroes&, int32_t, MP2::MapObjectType): Assertion `( world.GetTiles( tileIndex ).GetObject() == MP2::OBJ_CASTLE ) && Maps::isValidDirection( tileIndex, Direction::BOTTOM ) && hero.GetIndex() == Maps::GetDirectionIndex( tileIndex, Direction::BOTTOM )' failed.
Aborted (core dumped)
```
- fheroes2 1.0.7
- GOG Heroes of Might and Magic 2
- Fedora 39
- map Sharkania
- state reached by legitimate gameplay

**Comment by vlk-charles** [comment]
Just tested the latest snapshot (not release) build (commit `7b409`) and it does not crash anymore. So fixed?

**Comment by oleg-derevenetz** [comment]
Should be already fixed by #7567. Please try the latest snapshot.

**Comment by oleg-derevenetz** [comment]
> So fixed?

Yes, this issue should be fixed already some time ago.

**Comment by vlk-charles** [comment]
> Should be already fixed by #7567. Please try the latest snapshot.

Yeah, just found that too. I was previously only looking for open issues. My apologies.
